lib/SIL/Utils/OwnershipUtils.cpp

```diff
@@ -83,6 +83,22 @@ bool swift::hasPointerEscape(BorrowedValue value) {
   return false;
 }
 
+bool swift::hasPointerEscape(SILValue value) {
+  if (auto borrowedValue = BorrowedValue(value)) {
+    return hasPointerEscape(borrowedValue);
+  }
+  for (auto use : value->getUses()) {
```

Comment:
Checking immediate uses only makes sense for an owned value. Do you want an assert here and document that `\p value` must introduce an OSSA lifetime (either BorrowedValue or owned value)

Reply:
Fixed.  And added a check in RedundantMoveValueElimination that the original value is owned.